R/init.R

```diff
@@ -151,6 +153,17 @@ init <- function(data,
       modules$datanames <- if (identical(modules$datanames, "all")) {
         datanames
       } else if (is.character(modules$datanames)) {
+        extra_datanames <- setdiff(modules$datanames, datanames)
+        if (length(extra_datanames)) {
+          stop(
+            sprintf(
+              "Module %s has datanames that are not available in a 'data':\n %s not in %s",
+              modules$label,
+              toString(extra_datanames),
+              toString(datanames)
+            )
+          )
+        }
         datanames_adjusted <- intersect(modules$datanames, datanames)
         include_parent_datanames(dataname = datanames_adjusted, join_keys = join_keys)
```

Comment:
This seems to be neither assigned nor returned.

Reply:
yes it is. This is the code: 

```r
      modules$datanames <- if (identical(modules$datanames, "all")) {
        datanames
      } else if (is.character(modules$datanames)) {
        extra_datanames <- setdiff(modules$datanames, datanames)
        if (length(extra_datanames)) {
          stop(
            sprintf(
              "Module %s has datanames that are not available in a 'data':\n %s not in %s",
              modules$label,
              toString(extra_datanames),
              toString(datanames)
            )
          )
        }
        datanames_adjusted <- intersect(modules$datanames, datanames)
        include_parent_datanames(dataname = datanames_adjusted, join_keys = join_keys)
      }

``

Reply:
Right!